designer/server/src/main/scala/pl/touk/nussknacker/ui/process/migrate/TestModelMigrations.scala

```diff
@@ -91,6 +95,22 @@ class TestModelMigrations(
 
   }
 
+  private def processInParallel(
+      input: List[MigratedProcessDetails],
+      maxParallelism: Int
+  )(process: MigratedProcessDetails => TestMigrationResult): List[TestMigrationResult] = {
+    val pool        = new ForkJoinPool(maxParallelism)
```

Comment:
Is it a good idea to create a pool for every request? @piotrp wdyt? Maybe we should Move this pool outside to the place where we know if it is a good idea or not - to the place of `testMigrations` invocation

Reply:
I'm not sure, right now `testMigrations` is simple to use - you just pass it an additional integer (which we may redefine to `maxParallelism: Int = 1` to make it even simpler). I think migrations are heavy enough so that creating a thread pool here is justified, especially that this method is meant for performing operations in bulk.

Reply:
An idea: maybe let's do it Scala-way, and reuse `ExecutionContext` that is already being passed to `HttpRemoteEnvironment`? It's currently unused. Then, in cases where we don't care about degree of parallelism, we can just pass default EC.

Reply:
I updated the PR with following changes:

- I removed unused ExecutionContext from HttpRemoteEnvironment
- I modified testMigrations to use ExecutionContext instead of ForkJoinPool directly
- I parallelized fetchGroupByGroup operation
- I created custom ExecutionContext with thread prefixes for easier time with thread dumps

After @piotrp comment I will try to utilize previously unused ExecutionContext that I deleted from HttpRemoteEnvironment instead of creating one for collections operations.

Reply:
Don't restore EC to HttpRemoteEnvironment, it's enough that you have one passed to testMigration

Reply:
Ok, I added a commit where I use EC which is already passed to the method implicitly.